code/ATMOSPHERICS/components/binary_devices/passive_gate.dm

```diff
@@ -90,6 +90,13 @@
 	radio_connection.post_signal(src, signal, filter = RADIO_ATMOSIA)
 
 	return 1
+	
+/obj/machinery/atmospherics/binary/passive_gate/buildFrom(var/mob/usr,var/obj/item/pipe/pipe)
```

Comment:
we don't really have coding standards, but if we did, how would be best to do this in a less shit way than overriding 15 procs with the same copypasted code?
the parent atmospherics `unary/binary/ternary`?
the parent `atmospherics`?

Reply:
yeah i was thinking about what to do, but went with the copy paste method because its the same way they did the set_frequency proc

Reply:
pedr is right here; you should just make buildfrom at a higher level and you can simply add a sanity check to see if frequency is settable for that device and return early if not

bad earlier implementation isn't a good reason to continue that in the future

Reply:
also: do not call your arg "usr" which is a special name